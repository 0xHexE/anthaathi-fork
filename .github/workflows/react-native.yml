name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
