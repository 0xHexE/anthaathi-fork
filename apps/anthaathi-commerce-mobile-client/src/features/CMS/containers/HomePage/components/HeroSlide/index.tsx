import {View, GestureResponderEvent, ImageBackground} from 'react-native';
import React from 'react';
import {Text, useTheme, TouchableRipple} from 'react-native-paper';
import {MD3Colors} from 'react-native-paper/lib/typescript/types';
import {HomePageComponentType} from '../../../../types/common';

export interface HeroSlideProps {
  backgroundImageSrc: string;
  title: string;
  subTitle: string;
  buttonTitle: string;
  handlePress?: () => void;
}

const HeroSlide = (props: HeroSlideProps) => {
  const theme = useTheme();

  return (
    <View style={{marginVertical: 10}} testID="heroSlide">
      <ImageBackground
        testID="backgroundImage"
        source={{
          uri: props.backgroundImageSrc,
        }}
        style={{
          height: 240,
          width: '100%',
        }}>
        <View
          style={{
            height: '100%',
            width: '100%',
            backgroundColor: 'rgba(0,0,0,0.1)',
          }}>
          <View
            style={{
              flexGrow: 1,
              display: 'flex',
              alignItems: 'flex-start',
              justifyContent: 'flex-end',
              marginHorizontal: 20,
              marginVertical: 30,
            }}>
            <Text
              style={{
                color: (theme.colors as MD3Colors).background,
                fontSize: 24,
                fontWeight: 'bold',
                marginVertical: 5,
                paddingTop: 10,
              }}
              variant="titleSmall">
              {props.title}
            </Text>

            <Text
              style={{
                color: (theme.colors as MD3Colors).background,
                fontSize: 16,
                fontWeight: '400',
                marginVertical: 10,
              }}
              variant="titleSmall">
              {props.subTitle}
            </Text>
            {props.buttonTitle && (
              <HeroSlideButton
                label={props.buttonTitle}
                onPress={props.handlePress}
              />
            )}
          </View>
        </View>
      </ImageBackground>
    </View>
  );
};

function HeroSlideButton(props: {
  onPress: ((e: GestureResponderEvent) => void) | undefined;
  label: string;
}) {
  const theme = useTheme();

  return (
    <TouchableRipple testID="heroSlideOnPress" onPress={props.onPress}>
      <View
        style={{
          backgroundColor: (theme.colors as MD3Colors).background,
          borderRadius: 1,
        }}>
        <Text
          testID="buttonTitle"
          style={{
            color: (theme.colors as MD3Colors).onBackground,
            paddingLeft: 18,
            paddingRight: 18,
            paddingTop: 8,
            paddingBottom: 8,
            fontWeight: '700',
          }}
          variant="labelSmall">
          {props.label}
        </Text>
      </View>
    </TouchableRipple>
  );
}

export default HeroSlide;

export const HeroSlideCMSInput = {
  _component: HomePageComponentType.HeroSlide,
  component: HeroSlide,
};
