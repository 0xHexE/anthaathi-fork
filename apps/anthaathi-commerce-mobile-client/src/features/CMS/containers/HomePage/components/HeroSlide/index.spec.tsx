import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import HeroSlide from './index';

describe('HeroSlide', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <HeroSlide
            backgroundImageSrc="https://cdn.shopify.com/s/files/1/0648/1303/9842/files/fresh-squeezed-orange-juice_300x.jpg?v=1653584430"
            title="test"
            subTitle="test"
            buttonTitle="View All"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('heroSlide')).toBeTruthy();
  });

  it('should have HeroSlide title, subTitle', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <HeroSlide
            backgroundImageSrc="https://cdn.shopify.com/s/files/1/0648/1303/9842/files/fresh-squeezed-orange-juice_300x.jpg?v=1653584430"
            title="test title"
            subTitle="test subtitle"
            buttonTitle="View All"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('test title')).toBeTruthy();
    expect(temp.queryByText('test subtitle')).toBeTruthy();
  });

  it('should when we call tap', function () {
    const onpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <HeroSlide
            backgroundImageSrc="https://cdn.shopify.com/s/files/1/0648/1303/9842/files/fresh-squeezed-orange-juice_300x.jpg?v=1653584430"
            title="test title"
            subTitle="test subtitle"
            buttonTitle="View All"
            handlePress={onpress}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('heroSlideOnPress')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have HeroSlide Image', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <HeroSlide
            backgroundImageSrc="https://cdn.shopify.com/s/files/1/0648/1303/9842/files/fresh-squeezed-orange-juice_300x.jpg?v=1653584430"
            title="test title"
            subTitle="test subtitle"
            buttonTitle="View All"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('backgroundImage')!.props.source).toMatchObject({
      uri: 'https://cdn.shopify.com/s/files/1/0648/1303/9842/files/fresh-squeezed-orange-juice_300x.jpg?v=1653584430',
    });
  });

  it('should have HeroSlide button title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <HeroSlide
            backgroundImageSrc="https://cdn.shopify.com/s/files/1/0648/1303/9842/files/fresh-squeezed-orange-juice_300x.jpg?v=1653584430"
            title="test title"
            subTitle="test subtitle"
            buttonTitle="View All"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('buttonTitle')?.children).toContain('View All');
  });
});
