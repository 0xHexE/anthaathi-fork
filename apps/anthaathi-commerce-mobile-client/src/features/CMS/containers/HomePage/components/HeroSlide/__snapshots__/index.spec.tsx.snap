// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSlide should render the item 1`] = `
<View
  style={
    Object {
      "marginVertical": 10,
    }
  }
  testID="heroSlide"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": 240,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn.shopify.com/s/files/1/0648/1303/9842/files/fresh-squeezed-orange-juice_300x.jpg?v=1653584430",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 240,
            "width": "100%",
          },
          undefined,
        ]
      }
      testID="backgroundImage"
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.1)",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexGrow": 1,
            "justifyContent": "flex-end",
            "marginHorizontal": 20,
            "marginVertical": 30,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "rgba(255, 251, 254, 1)",
                "fontSize": 24,
                "fontWeight": "bold",
                "marginVertical": 5,
                "paddingTop": 10,
              },
            ]
          }
        >
          test
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "rgba(255, 251, 254, 1)",
                "fontSize": 16,
                "fontWeight": "400",
                "marginVertical": 10,
              },
            ]
          }
        >
          test
        </Text>
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              undefined,
            ]
          }
          testID="heroSlideOnPress"
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(255, 251, 254, 1)",
                "borderRadius": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontSize": 11,
                    "fontWeight": "500",
                    "letterSpacing": 0.5,
                    "lineHeight": 16,
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontWeight": "700",
                    "paddingBottom": 8,
                    "paddingLeft": 18,
                    "paddingRight": 18,
                    "paddingTop": 8,
                  },
                ]
              }
              testID="buttonTitle"
            >
              View All
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
